File: exp8.1.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cout << "Enter the number of Processes : ";
    cin >> n;
    int bt[n], wt[n], tat[n], avwt = 0, avtat = 0;
    cout << "Enter Process Burst Time for each process : ";
    for (int i = 0; i < n; i++)
    {
        cin >> bt[i];
    }
    wt[0] = 0;
    for (int i = 1; i < n; i++)
    {
        wt[i] = 0;
        for (int j = 0; j < i; j++)
        {
            wt[i] += bt[j];
        }
    }

    cout << "Process\tBurst Time\tWaiting Time\tTurnaround Time" << endl;
    for (int i = 0; i < n; i++)
    {
        tat[i] = bt[i] + wt[i];
        avwt += wt[i];
        avtat += tat[i];
        cout << i + 1 << "\t" << bt[i] << "\t\t" << wt[i] << "\t\t" << tat[i] << endl;
    }
    avwt /= n;
    avtat /= n;
    cout << "Average Waiting Time\t" << avwt << endl;
    cout << "Average Turnaround Time\t" << avtat << endl;

    return 0;
}

File: exp8.1.py
numProc = int(input("Enter number of processes: "))

print("Enter burst time for each process: ")

burst = [int(input()) for i in range (numProc)]

wait = [0]

for i in range(1,numProc):
    wait.append(0)
    for j in range(i):
        wait[i] += burst[j]

tat = []

print("Process\tBurst Time\tWaiting Time\tTurnaround Time")
for i in range (numProc):
    tat.append(burst[i] + wait[i])
    print(i,"\t",burst[i],"\t\t",wait[i],"\t\t",tat[i])

print("Average waiting time:",sum(wait)/numProc)
print("Average turnaround time:",sum(tat)/numProc)

